### PR TITLE
devenv/ci(feat): Add shellcheck to the devenv & CI

### PR DESCRIPTION
**Pull request by armenzg**
This will help catch issues in the bash file.

### CODE REVIEW

File: .github/workflows/linter.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  pull_request:
+
+jobs:
+  shellcheck:
+    name: Shellcheck
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run ShellCheck
+      uses: ludeeus/action-shellcheck@master
```
Comment (billyvg) [inline .github/workflows/linter.yml:15]: Can we fork or pin to a SHA?

File: .github/workflows/linter.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  pull_request:
+
+jobs:
+  shellcheck:
+    name: Shellcheck
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run ShellCheck
+      uses: ludeeus/action-shellcheck@1.0.0
```
Comment (billyvg) [inline .github/workflows/linter.yml:15]: Version is not enough, it needs to be a SHA: https://www.notion.so/sentry/GitHub-Actions-Guidelines-ade10a2ff0364562bd248ae92163a524

Comment (armenzg) [reply .github/workflows/linter.yml]: Right! Thank you.

**Review by billyvg — APPROVED** [review]

File: bootstrap.sh [hunk 3]
```diff
@@ -279,11 +279,11 @@ RUBY
 # Check and install any remaining software updates.
 software_update() {
   logn "Checking for software updates:"
-  if softwareupdate -l 2>&1 | grep $Q "No new software available."; then
+  if softwareupdate -l 2>&1 | grep "$Q" "No new software available."; then
     logk
   else
     echo
-    if [ $1 == "reminder "]; then
```
Comment (armenzg) [inline bootstrap.sh:286]: This extra space was actually a bug in the code.

**Pull request merged**